compiler/rustc_lint/src/builtin.rs

```diff
@@ -1228,6 +1228,8 @@ impl<'tcx> LateLintPass<'tcx> for TrivialConstraints {
                     Subtype(..) |
                     ConstEvaluatable(..) |
                     ConstEquate(..) => continue,
+
+                    TypeFromEnv(..) => unimplemented!(),
```

Comment:
Should this variant be ignored?

Reply:
Yep, just include it above and `continue`